src/rule_evaluator.ts

```diff
@@ -40,24 +55,60 @@ function evaluateCondition(subjectAttributes: Record<string, any>, condition: Co
       case OperatorType.MATCHES:
         return new RegExp(condition.value as string).test(value as string);
       case OperatorType.ONE_OF:
-        return isOneOf(value, condition.value);
+        return isOneOf(
+          value.toString().toLowerCase(),
+          condition.value.map((value: string) => value.toLowerCase()),
+        );
       case OperatorType.NOT_ONE_OF:
-        return isNotOneOf(value, condition.value);
+        return isNotOneOf(
+          value.toString().toLowerCase(),
+          condition.value.map((value: string) => value.toLowerCase()),
+        );
+    }
+  }
+  return false;
+}
+
+function evaluateObfuscatedCondition(
+  subjectAttributes: Record<string, any>,
+  condition: Condition,
+): boolean {
+  const hashedSubjectAttributes: Record<string, any> = Object.entries(subjectAttributes).reduce(
+    (accum, [key, val]) => ({ [getMD5Hash(key)]: val, ...accum }),
+    {},
+  );
+  const value = hashedSubjectAttributes[condition.attribute];
+  if (value != null) {
+    switch (condition.operator) {
+      case getMD5Hash(OperatorType.GTE):
+        return compareNumber(value, Number(decodeBase64Hash(condition.value)), (a, b) => a >= b);
+      case getMD5Hash(OperatorType.GT):
+        return compareNumber(value, Number(decodeBase64Hash(condition.value)), (a, b) => a > b);
+      case getMD5Hash(OperatorType.LTE):
+        return compareNumber(value, Number(decodeBase64Hash(condition.value)), (a, b) => a <= b);
+      case getMD5Hash(OperatorType.LT):
+        return compareNumber(value, Number(decodeBase64Hash(condition.value)), (a, b) => a < b);
+      case getMD5Hash(OperatorType.MATCHES):
+        return new RegExp(decodeBase64Hash(condition.value)).test(value as string);
+      case getMD5Hash(OperatorType.ONE_OF):
+        return isOneOf(getMD5Hash(value.toString().toLowerCase()), condition.value);
+      case getMD5Hash(OperatorType.NOT_ONE_OF):
+        return isNotOneOf(getMD5Hash(value.toString().toLowerCase()), condition.value);
     }
   }
   return false;
 }
 
-function isOneOf(attributeValue: any, conditionValue: string[]) {
-  return getMatchingStringValues(attributeValue.toString(), conditionValue).length > 0;
+function isOneOf(attributeValue: string, conditionValue: string[]) {
+  return getMatchingStringValues(attributeValue, conditionValue).length > 0;
 }
 
-function isNotOneOf(attributeValue: any, conditionValue: string[]) {
-  return getMatchingStringValues(attributeValue.toString(), conditionValue).length === 0;
+function isNotOneOf(attributeValue: string, conditionValue: string[]) {
+  return getMatchingStringValues(attributeValue, conditionValue).length === 0;
 }
 
 function getMatchingStringValues(attributeValue: string, conditionValues: string[]): string[] {
-  return conditionValues.filter((value) => value.toLowerCase() === attributeValue.toLowerCase());
```

Comment:
Moved the lower-casing further up so that we can continue to do case-insensitive matching for both the obfuscated and un-obfuscated RACs. If we are matching on hashed values, we need to lowercase the value before hashing it